src/components/NavBar/index.tsx

```diff
@@ -41,17 +41,7 @@ const NavBar = () => {
   return (
     <AppBar className="navbar">
       <div className="navbar-div">
-        <Link to={routes.HOME} className="navbar-info" onClick={reloadHomePage}>
-          <img src={logo} alt="logo" className="logo" />
-          <div className="navbar-text">
-            <span className="navbar-company">
-              <Typography>ProductDock</Typography>
-            </span>
-            <span className="navbar-title">
-              <Typography>Library</Typography>
-            </span>
-          </div>
-        </Link>
+        <NavBarLogo handleClickLogo={reloadHomePage} />
```

Comment:
Great, we just need to do some fine-tuning here. Navigation can be fully encapsulated within NavBarLogo. Parent components (like NavBar or AdminNavBar) don't need to know how to implement navigation. What they know already is what "Home" means in their case. NavBar knows its home is "/" and AdminNavBar knows its home is "/admin". You can make use of that, by just passing the home route to the <NavBarLogo homeRoute=...> and letting NavBarLogo handle the routing. 

There will be one more problem to solve - in the case of Readers nav bar clicking the home button also resets search parameters. Can you find a way to include this without making NavBarLogo aware of the use case (is it for admins or for readers)?